src/OmniSharp.Host/Services/AssemblyLoader.cs

```diff
@@ -38,17 +38,26 @@ public IReadOnlyList<Assembly> LoadAllFrom(string folderPath)
         {
             if (string.IsNullOrWhiteSpace(folderPath)) return Array.Empty<Assembly>();
 
-            var assemblies = new List<Assembly>();
-            foreach (var filePath in Directory.EnumerateFiles(folderPath, "*.dll"))
+            try
             {
-                var assembly = LoadFromPath(filePath);
-                if (assembly != null)
+                var assemblies = new List<Assembly>();
+
+                foreach (var filePath in Directory.EnumerateFiles(folderPath, "*.dll"))
                 {
-                    assemblies.Add(assembly);
+                    var assembly = LoadFromPath(filePath);
+                    if (assembly != null)
+                    {
+                        assemblies.Add(assembly);
+                    }
                 }
-            }
 
-            return assemblies;
+                return assemblies;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"An error occurred when attempting to access '{folderPath}'.");
```

Comment:
This message could be wrong if something else fails. I can't see what would fail though. :smile:

Is there any reason not to check `Directory.Exists(folderPath)`?